### PR TITLE
Avoid requiring atomics on quickjs-libc

### PR DESCRIPTION
**Pull request by saghul**

### CODE REVIEW

**Comment by saghul** [comment]
@bnoordhuis PTAL, not sure if TSAN will be happy, let's see.

**Comment by saghul** [comment]
TSan run: https://github.com/quickjs-ng/quickjs/actions/runs/11454131351

**Comment by saghul** [comment]
All good! WDYT Ben? I think it's kinda nice to put it there in the runtime state.

File: quickjs-libc.c [hunk 1]
```diff
@@ -162,14 +162,14 @@ typedef struct JSThreadState {
     int eval_script_recurse; /* only used in the main thread */
     int64_t next_timer_id; /* for setTimeout / setInterval */
     JSValue exc; /* current exception from one of our handlers */
+    int can_js_os_poll;
```
Comment (bnoordhuis) [inline quickjs-libc.c]:
```suggestion
    BOOL can_js_os_poll;
```

**Review by bnoordhuis — APPROVED** [review]
Now do `os_pending_signals` next :-)

**Pull request merged**